cli/maintainer/src/maintenance.rs

```diff
@@ -754,33 +772,75 @@ impl SolidoState {
             .exchange_sol(Lamports(anker_state.b_sol_total_supply_amount.0))
             .expect("It will not overflow because we always have less than the total amount of minted Sol.");
 
-        let rewards = (reserve_st_sol - st_sol_amount).ok()?;
+        (reserve_st_sol - st_sol_amount).ok()
+    }
+
+    /// Try to sell the extra stSOL rewards for UST tokens or
+    /// to update the historical pool price exchange rate to protect us
+    /// against sandwiching attacks.
+    pub fn try_sell_anker_rewards(&self) -> Option<MaintenanceInstruction> {
+        let anker_state = self.anker_state.as_ref()?;
 
+        let rewards = self.get_anker_rewards()?;
         let min_rewards_to_sell = self
             .solido
             .exchange_rate
             .exchange_sol(Self::MINIMUM_WITHDRAW_AMOUNT)
             .expect("The price of a signature should be small enough that it doesn't overflow.");
 
-        // Estimate the amount of UST that we will get when swapping against the constant product
-        // pool. This is only an approximation because it doesn’t uphold the constant product
-        // invariant, but this is a reasonable approximation when the rewards to sell are small
-        // compared to the pool balance. Also, this ignores swap fees. It would have been nice
-        // to use the actual `spl_token_swap::state::SwapV1` instance, but it is not `Send`, so
-        // we can’t store it in our state.
```

Comment:
I think this comment was useful for understanding why we do `rewards * expected_st_sol_price_of_one_ust` instead of computing the more precise value from the `SwapV1` instance, can we preserve it?

Reply:
Humm I see, My idea was to use the curve calculator so we don't do the `rewards * expected_st_sol_price_of_one_ust`, I think I forgot to include it. I'll do it.